x-pack/plugins/spaces/server/routes/api/external/get_shareable_references.ts

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { schema } from '@kbn/config-schema';
+
+// import { tagSavedObjectTypeName } from '../../../../../saved_objects_tagging/common';
+import type { GetShareableReferencesResponse } from '../../../../common';
+import { UNKNOWN_SPACE } from '../../../../common/constants';
+import { wrapError } from '../../../lib/errors';
+import { createLicensedRouteHandler } from '../../lib';
+import type { ExternalRouteDeps } from './';
+
+const tagSavedObjectTypeName = 'tag'; // TODO: import this from saved_objects_tagging plugin (cannot import directly, generates TS circular graph error)
+
+export function initGetShareableReferencesApi(deps: ExternalRouteDeps) {
+  const { externalRouter, getStartServices } = deps;
+
+  externalRouter.post(
+    {
+      path: '/api/spaces/_get_shareable_references',
+      validate: {
+        body: schema.object({
+          objects: schema.arrayOf(schema.object({ type: schema.string(), id: schema.string() })),
+        }),
+      },
+    },
+    createLicensedRouteHandler(async (context, request, response) => {
+      const [startServices] = await getStartServices();
+      const scopedClient = startServices.savedObjects.getScopedClient(request);
+
+      const { objects } = request.body;
+
+      try {
+        const collectedObjects = await scopedClient.collectMultiNamespaceReferences(objects, {
+          purpose: 'updateObjectsSpaces',
+        });
+        const requestedObjectsSet = objects.reduce(
+          (acc, { type, id }) => acc.add(`${type}:${id}`),
+          new Set<string>()
+        );
```

Comment:
Not my area, but I feel like this logic should be extracted out of the handler to its own utility function.

Reply:
I agree, I'll do it.